packages/core/src/components/menu/_menu.scss

```diff
@@ -229,27 +229,31 @@ Styleguide components.menu.css
     color: $pt-text-color-muted;
   }
 
-  &:not(.pt-disabled):hover {
-    &::before,
-    &::after,
-    .pt-menu-item-label {
-      color: $white;
-    }
-
+  &:not(.pt-disabled):not(.pt-active):hover {
     // intent color only appears on hover
     @each $intent, $color in $pt-intent-colors {
       &.pt-intent-#{$intent} {
         background-color: $color;
+        color: $white;
+
```

Comment:
We should handle the `:active` state for these intents, maybe worth having a `menu-item-intent` mixin, similarly to buttons?